### PR TITLE
Add localization system (gettext)

### PR DESCRIPTION
**Pull request by Ansa89**

### CODE REVIEW

File: translations/toxic.pot [hunk 1]
```diff
@@ -0,0 +1,1574 @@
+# SOME DESCRIPTIVE TITLE.
```
Comment (JFreegman) [inline translations/toxic.pot]: Should probably put the real license here

**Comment by JFreegman** [comment]
When I said I'd merge it with my new groupchats branch I lied. That would be way too tedious. Either you should remove any changes to groupchat.c, groupchat.h and group_commands.c, or try to merge it yourself.

**Comment by Ansa89** [comment]
> That would be way too tedious.

Why? If you merge both PR (Tox/toxic#337 and JFreegman/toxic#6) then there shouldn't be too merge conflicts.

**Comment by JFreegman** [comment]
I didn't even notice that you made a pr on that branch, my bad.

**Comment by JFreegman** [comment]
@Ansa89 how well tested is this?

**Comment by Ansa89** [comment]
@JFreegman it works on my pc, but I don't know if anybody else tested this (probably you need some tests for OSX).

**Comment by JFreegman** [comment]
I just need confirmation that it doesn't break anything on any supported platform then I'll merge it. I can't test on OSX myself.

**Comment by stal888** [comment]
@JFreegman there is a ~78% chance this will work on OS X.

I didn't actually test because the makefile sucks, sorry.

**Comment by Ansa89** [comment]
@JFreegman if something doesn't work, you can always use `make DISABLE_LOCALIZATION=1` to come back to the old behavior.

For OSX user I updated the INSTALL.md adding gettext as dependency to be installed through brew.

**Comment by JFreegman** [comment]
@Ansa89 I'm having a lot of trouble merging this with my new groupchat branch. I merged your separate pull request, but now there are a ton of new merge conflicts when I try to merge with the master branch. These aren't normal conflicts either, but hundreds/thousands of conflicts in the mo and po files. I tried updating and rebuilding the translations to no avail.

Also, am I missing something here or are you supposed to update the translations every time you make a change in the source code?

**Comment by JFreegman** [comment]
I'm going to have to revert this if it doesn't get resolved soon because it's preventing me from moving forward with the new groupchats branch. I'm hoping that I'm just stupid and missing something obvious, but if this breaks the ability to merge branches then it's not a viable solution.

**Comment by louipc** [comment]
On Tue 30 Jun 2015 20:57 -0700, JFreegman wrote:
> @Ansa89 I'm having a lot of trouble merging this with my new groupchat
> branch. I merged your separate pull request, but now there are a ton of new
> merge conflicts when I try to merge with the master branch. These aren't
> normal conflicts either, but hundreds/thousands of conflicts in the mo and po
> files. I tried updating and rebuilding the translations to no avail.
> 
> Also, am I missing something here or are you supposed to update the
> translations every time you make a change in the source code?

I think the best time to update translations would be some time before a new
release. Then put the word out and give some time for translators to do their
updates. This is called a 'string freeze'.

From what I can tell the problem is that translations and updates and groupchat
features are interleaved between two branches which is now a big messy hairball.
This will be a big job to clean up.

Translation updates should never have entered the new_groupchats branch.
That should have made rebasing new_groupchats onto master a lot easier.

Yeah there is going to have to be a lot of resetting, rebasing, and/or
reverting unfortunately. The exact approach will depend on you as the
maintainer.

In the future, people should rebase their topic branch on master to include new
features and to help avoid conflicts. If someone wants their branch merged they
need do the work to resolve the conflicts themselves. It'll make your life
a lot easier.

**Comment by JFreegman** [comment]
>In the future, people should rebase their topic branch on master to include new
features and to help avoid conflicts. If someone wants their branch merged they
need do the work to resolve the conflicts themselves. It'll make your life
a lot easier.

This makes my life easier, but it makes toxic development in general much more difficult. There must be an easier/automated solution, but it's not something I have time to research right now.

**Comment by louipc** [comment]
On Thu 02 Jul 2015 14:21 -0700, JFreegman wrote:
> This makes my life easier, but it makes toxic development in general much
> more difficult. There must be an easier/automated solution, but it's not
> something I have time to research right now.

Unfortunately, we don't have sophisticated autonomous programmer robots that
make judgements regarding conflicts. Maybe in a couple of years.

Until then, humans will need to learn how to use the tools and to submit and
accept proper changesets.

You can try git-rerere to aid in the task or conflict resolution.
Git also allows you to define your own merge driver, which is really getting
into the guts of git. This is where you would put your autobot.
https://www.kernel.org/pub/software/scm/git/docs/gitattributes.html#_defining_a_custom_merge_driver

I have never used either. I'm not that fancy though.

Merge strategies is a way in git to prefer one changeset over another (theirs
vs ours) during merging to resolve conflicts. In my experience the preferred
change is a combination of both and that needs to be done manually.

Honestly I'm pretty good with git, not a super expert, but from what I see you
will need to 'start over' either with the i18n patchset or the groupchat
patchset and keep the hairball(s) as reference. Redoing the groupchat patches
will be the path of least pain since it hasn't been merged to master yet.

**Comment by JFreegman** [comment]
I just reverted the merge. I would love to see toxic supporting multiple languages, but I just don't have the time to deal with any of this myself. My current priority is with the new groupchats, so for the time being, all I'm here for is merging and making sure things work smoothly.

**Pull request merged**